test/unit/range/view/view_take_line_test.cpp

```diff
@@ -159,8 +159,7 @@ TEST(view_take_line, reverse_bug)
     EXPECT_TRUE(const_iterable_concept<decltype(v1)>);
     EXPECT_TRUE((std::ranges::OutputRange<decltype(v1), char>));
 
-    // Either of these build-fail on GCC7 and GCC8
-//     auto v2 = v1 | ranges::view::reverse;
-//     auto v2 = ranges::reverse_view{v1};
+    // No build failure, but wrong results:
```

Comment:
👀 What's the result?

Reply:
The result is the whole underlying range reversed, instead of the sub-string. I tried to debug it for an hour, but didn't come up with a fix for range-v3. Working around the issue by using a distinct sentinel and storing end in the iterator, leads to problems with GCC. I will open a card for it.

Context: The problem is that range-v3 decrements the end iterator if possible to go to the last element, but it should only do so if the range is a CommonRange.